test/extensions/filters/http/oauth2/oauth_integration_test.cc

```diff
@@ -122,6 +125,73 @@ name: oauth
 
     HttpIntegrationTest::initialize();
   }
+
+  void validateHmac(const Http::ResponseHeaderMap& headers, absl::string_view host,
+                    absl::string_view hmac_secret) {
+    std::string expires = Http::Utility::parseSetCookieValue(headers, "OauthExpires");
+    std::string token = Http::Utility::parseSetCookieValue(headers, "BearerToken");
+    std::string hmac = Http::Utility::parseSetCookieValue(headers, "OauthHMAC");
+
+    Http::TestRequestHeaderMapImpl validate_headers{{":authority", std::string(host)}};
+
+    validate_headers.addReferenceKey(Http::Headers::get().Cookie, absl::StrCat("OauthHMAC=", hmac));
+    validate_headers.addReferenceKey(Http::Headers::get().Cookie,
+                                     absl::StrCat("OauthExpires=", expires));
+    validate_headers.addReferenceKey(Http::Headers::get().Cookie,
+                                     absl::StrCat("BearerToken=", token));
+
+    OAuth2CookieValidator validator{api_->timeSource()};
+    validator.setParams(validate_headers, std::string(hmac_secret));
+    ASSERT_TRUE(validator.isValid());
```

Comment:
if you return `validator.isValid()` here instead you can do an `EXPECT_TRUE` at call site and provide better error messaging